packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -347,67 +347,32 @@ const positiveNegativeNumberCellView = PositiveNegativeNumberCellView.compose({
 });
 ```
 
-Finally, here is a column element that allows a user to register a callback for a click event on a button inside the cell template:
+### Event Delegation
 
-```TS
-type TableColumnButtonCellData = TableStringField<'id'>;
-
-public class TableColumnButton extends TableColumn<TableColumnButtonCellData> {
-    ...
-
-    public cellRecordFieldNames = ['id'] as const;
+Because the cell view is not a descendant of the column element, we must add special support for the column to handle events that originate in the cells. A column will define a `columnDelegatedEvents` property which is an array of event names that should be delegated.
 
-    @attr
-    public idKey: string;
-
-    public getDataRecordFieldNames(): string[] {
-        return [valueKey];
-    }
-
-    public cellViewTag = 'table-cell-view-button';
-
-    public validateCellData(cellData: TCellRecord): void {
-        if (typeof(cellData['id']) !== 'string') {
-            throw new Error('Type for cellData is incorrect!');
-        }
-    }
+```TS
+TableColumn {
+   abstract readonly columnDelegatedEvents: readonly string[];
 }
 
-class ButtonCellView extends TableCellView<TableColumnButtonCellData> {
-    public onButtonClick(): void {
-        this.$emit('button-click', { data: this.cellRecord.id })
-    }
+AnchorTableColumn extends TableColumn {
+   readonly columnDelegatedEvents = ['click'] as const;
 }
-const buttonCellView = ButtonCellView.compose({
-    baseName: 'table-cell-view-button',
-    template: html<ButtonCellView>`
-        <nimble-button @click="${(x) => x.onButtonClick()}">
-            <span>Press Me</span>
-        </nimble-button>`,
-    styles: /* styling */
-});
-```
-
-Angular template:
-
-```HTML
-<nimble-table (button-click)="doSomething($event)">
-    <nimble-table-column-button></nimble-table-column-button>
-</nimble-table>
 ```
 
-Angular code:
+Upon connecting a cell view to the DOM, we will attach an event listener for each event type in `columnDelegatedEvents`. The handler will wrap the original event in a new `CustomEvent` and dispatch that to the column. It will also stop propagation on the original event, and prevent default if the `dispatchEvent` call returns `false`.
```

Comment:
> It will also stop propagation on the original event, and prevent default if the `dispatchEvent` call returns `false`

This seems like it is crossing wires unnecessarily. There are two events being issued:
- 'delegated-events' (table column notifying that an event came from the view)
- the original event (the original event coming from the view)

So to check any state on the element for the original event I check the original event target. But to prevent default of the original event I don't call prevent default of the original event, instead I call prevent default on the delegated-event? Also nothing stops me from calling original event preventDefault, there are now just 2 ways to do the same thing.

Why not keep them completely separate? If I want to prevent default of the original event I just do that explicitly.

Reply:
Sure, those are good points. No longer calling `preventDefault()` here.